Self-Evolving-Car/Self-Evolving-Bot/Self-Evolving-Bot.py

```diff
@@ -0,0 +1,240 @@
+        car_to_ball = ball_location - car_location
+        self.distance_to_ball[self.frame] = math.sqrt(pow(my_car.physics.location.x-packet.game_ball.physics.location.x,2)+pow(my_car.physics.location.y-packet.game_ball.physics.location.y,2)+pow(my_car.physics.location.z-packet.game_ball.physics.location.z,2))
+        distance_to_ball_x = packet.game_ball.physics.location.x - my_car.physics.location.x
+        distance_to_ball_y = packet.game_ball.physics.location.y - my_car.physics.location.y
+        distance_to_ball_z = packet.game_ball.physics.location.z - my_car.physics.location.z
+
+
+        #RENDER RESULTS
+        action_display = "GEN: "+str(self.gen+1)+" | BOT: "+str(self.brain)#+" \nTURN: "+str(self.botList[self.brain].Nodes[1].node(self.distance_to_ball_x,self.distance_to_ball_y,self.distance_to_ball_z,4,5,6,7))+" \nBOOST: "+str(self.botList[self.brain].Nodes[2].node(self.distance_to_ball_x,self.distance_to_ball_y,self.distance_to_ball_z,8,9,10,11))+" \nTHROTTLE: "+str(self.botList[self.brain].Nodes(self.distance_to_ball_x,self.distance_to_ball_y,self.distance_to_ball_z,0,1,2,3))
+        if self.frame > 5000:
+            self.frame = 0
+        draw_debug(self.renderer, my_car, packet.game_ball, action_display)
+
+
+        #GAME STATE
+        car_state = CarState(boost_amount=100)
+        ball_state = BallState(Physics(velocity=Vector3(0, 0, 0), location=Vector3(0, -1000, 1200),angular_velocity=Vector3(0, 0, 0)))
+        game_state = GameState(ball=ball_state,cars={self.index: car_state})
+        self.set_game_state(game_state)
+
+
+        #NEURAL NET OUTPUTS
+        hidden1 = self.botList[self.brain].Nodes[0].node(distance_to_ball_x, distance_to_ball_y, distance_to_ball_z, 0, 1, 2, 3)
+        hidden2 = self.botList[self.brain].Nodes[1].node(distance_to_ball_x, distance_to_ball_y, distance_to_ball_z, 4, 5, 6, 7)
+        hidden3 = self.botList[self.brain].Nodes[2].node(distance_to_ball_x, distance_to_ball_y, distance_to_ball_z, 8, 9, 10, 11)
+        hidden4 = self.botList[self.brain].Nodes[3].node(distance_to_ball_x, distance_to_ball_y, distance_to_ball_z, 12, 13, 14, 15)
+
+        self.controller_state.pitch = self.botList[self.brain].Nodes[4].node(hidden1, hidden2, hidden3, hidden4, 16, 17, 18, 19)
+        self.controller_state.yaw = self.botList[self.brain].Nodes[5].node(hidden1, hidden2, hidden3, hidden4, 20, 21, 22, 23)
+        if self.botList[self.brain].Nodes[6].node(hidden1, hidden2, hidden3, hidden4, 24, 25, 26, 27) > 0: self.controller_state.boost = True
+        else: self.controller_state.boost = False
+
+
+        #KILL
+        if (my_car.physics.location.z < 100 or my_car.physics.location.z > 1950 or my_car.physics.location.x < -4000 or my_car.physics.location.x > 4000 or my_car.physics.location.y > 5000) and self.frame > 50:
+            #self.botList[self.brain].fitness = (1/self.frame+1)*100000
+            self.frame = 5000
+
+        return self.controller_state
+
+    def calcFitness(self):

```

Comment:
Move fitness outside of the agent file

Reply:
Sorry! Ill do that in my next commit

Reply:
I don't think this is essential for the pull request. Can we resolve this?